### DISCUSSION TITLE
How to use API rules when extending with JavaScript Routing?

### DISCUSSION THREAD

**Discussion by flashvnn** [pocketbase/pocketbase#5018] [Q&A] [answered]
i extending pocketbase with JavaScript .
i adding new routing and i want to use API rules inside the routing define.
Example:

```javascript
routerAdd("GET", "/hello/:name", (c) => {
    const validated = RulesValidate('@request.auth.id != "" && allowed_users.id ?= @request.auth.id');
    if (validated) {
      let name = c.pathParam("name")
      return c.json(200, { "message": "Hello " + name })
   } else {
    return c.json(400, { "message": "Access error"})
   }
})
```

This will help easy define access rule for custom route and i can define api rules in custom collection and change it later without change in the js code.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand the use case and the desired behavior but the only existing helper at the moment that allows checking a single record against an API rule is the `$app.dao().canAccessRecord()`:

```js
const info = $apis.requestInfo(c)
const canAccess = $app.dao().canAccessRecord(record, info, "your api rule....")
```
(_see also [Check if record can be accessed](https://pocketbase.io/docs/js-records/#check-if-record-can-be-accessed) example_)
